### PR TITLE
Adds python interface to Filter, MovingWindowFilter, RotationSpline.

### PR DESCRIPTION
**Pull request by LolaSegura**
# 🎉 New feature
Goes on top of #221
Related to #101 #210

## Summary
Adds Python interface for three math classes: Filter, MovingWindowFilter and RotationSPline. For each class a python test has been created.

### Filter
The `Filter.hh` file contains  various template classes that inherit from a `Filter` class, which is a template class itself. To create binding for this classes an instance of `Filter<type>` had to be declared previously. 
There were some classes that inherit from an instance of another class. For example `class OnePoleQuaternion : public OnePole<math::Quaterniond>`. This wasn't being interpreted by swig as expected. The inherited methods that were not  override, were not being created in the binding. The solution we implemented for this was define these classes inside the interface file without the inheritance and adding the methods that were not being created. A similar approach was taken in the `SignalStats` class (#220 ).

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

### CODE REVIEW

**Comment by francocipollone** [comment]
This PR is on hold until #221 is merged.

**Review by francocipollone — APPROVED** [review]
LGTM

**Comment by francocipollone** [comment]
I'll solve the conflicts after #221 is merged.

File: src/python/MovingWindowFilter.i
Comment (azeey) [inline src/python/MovingWindowFilter.i:24]: Removing this line fixes the compilation error for me. Maybe it's a known issue, but I had to `touch python.i` for it to recompile this file.

Comment (francocipollone) [reply src/python/MovingWindowFilter.i:24]: Let me take a look. It seemed to be working before merging this branch with ign-math6.

Comment (francocipollone) [reply src/python/MovingWindowFilter.i]: You were right. That import was causing issues because of the syntaxis: it should be `include` instead of `import`

The truth is that that import wasn't necessary given that `std::vector` isn't being used in the API, so that import statement wasn't needed at all. I pushed the fix

**Review by scpeters — APPROVED** [review]

**Pull request merged**
